drivers/hmis/app/models/hmis/ce/match/engine.rb

```diff
@@ -15,7 +15,9 @@ def self.call(...)
     # 2. Evaluate the eligibility requirement expression against each matched client. We expect all expression variables to be defined.
     def call(pool, clients)
       validate_clients_parameter!(clients)
-      # TODO: remove this in #7671
+      # TODO: remove this in #7671. store mapping of { destination_id => hmis_source_client_id }
```

Comment:
I'm confused about the mapping suggested in this comment -- My understanding is, in #7671,
- we want the engine to continue to receive a scope of source clients
- it should translate that into a scope of dest clients (per comment on line 20)
- client proxies should be instantiated with dest client IDs
- why do we need to map back from the dest to hmis source client ID?

Reply:
Oh, sorry, I was confused because I thought "store mapping of { destination_id => hmis_source_client_id }" was an instruction for the todo associated with #7671. But I guess it isn't, it is a description of what line 19 is doing currently. Same with comment on line 20. Both comments are _both_ adding a TODO for future removal and describing current behavior.